crates/red_knot_python_semantic/resources/mdtest/annotations/any.md

```diff
@@ -0,0 +1,68 @@
+# Any
+
+## Annotation
+
+`typing.Any` is a way to name the Any type.
+
+```py
+from typing import Any
+
+x: Any = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Aliased to a different name
+
+If you alias `typing.Any` to another name, we still recognize that as a spelling of the Any type.
+
+```py
+from typing import Any as RenamedAny
+
+x: RenamedAny = 1
+x = "foo"
+
+def f():
+    reveal_type(x)  # revealed: Any
+```
+
+## Shadowed class
+
+If you define your own class named `Any`, using that in a type expression refers to your class, and
+isn't a spelling of the Any type.
+
+```py
+class Any:
+    pass
+
+x: Any
+
+def f():
+    reveal_type(x)  # revealed: Any
+
+# This verifies that we're not accidentally seeing typing.Any, since str is assignable
+# to that but not to our locally defined class.
+y: Any = "not an Any"  # error: [invalid-assignment]
+```
+
+## Subclass
+
+The spec allows you to define subclasses of `Any`, which must also resolve to the Any type.
+
+```py
+from typing import Any
+
+class Subclass(Any):
+    pass
+
+# Since Subclass is a subclass of Any, it is assignable to and from any other type, just like Any.
+x: Subclass = 1
+y: int = Subclass()
```

Comment:
I think I've updated the inference logic correctly to treat instances of `Subclass` as if they were `Any` instead of `Instance(Subclass)`.  These lines both got assignment errors before the change.

Reply:
I don't think we want to make that change; `Subclass` should still have the type `Instance(Subclass)`, not the type `Any`. Our representation of the class `Subclass` will have `Any` in its method resolution order (MRO), which will impact how we treat it in some cases. For example, it should be assignable to any other non-final type, since it might be a subclass of that type, as mentioned below, and accessing an unknown attribute on it should give an attribute type of `Any` (the unknown base type might have that attribute, with some unknown type). But these are cases that we will implement in e.g. `Type::is_assignable_to` and `Type::member`, not by making the subclass of Any equivalent to Any.